sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/OffsetSeqLog.scala

```diff
@@ -46,6 +47,23 @@ import org.apache.spark.sql.connector.read.streaming.{Offset => OffsetV2}
 class OffsetSeqLog(sparkSession: SparkSession, path: String)
   extends HDFSMetadataLog[OffsetSeq](sparkSession, path) {
 
+  private val cachedMetadata = new ju.TreeMap[Long, OffsetSeq]()
```

Comment:
What about moving `cachedMetadata` to  `HDFSMetadataLog`? Seems like other metadata logs can benefit too.

Reply:
There're different usage patterns and characteristics against these metadata logs, and the change is rather only bound to the usage pattern of offsetLog.
e.g. commitLog is only added per batch, and read only when query is started or restored from checkpoint.

Reply:
@zsxwing thoughts?

Reply:
You can check by yourself: find `commitLog.get(` and you get nothing except tests. find `commitLog.getLatest(` and realize that is only happening when starting/restarting query. Simple enough?